crates/ruff_linter/src/rules/refurb/rules/subclass_builtin.rs

```diff
@@ -9,21 +9,34 @@ use crate::{checkers::ast::Checker, importer::ImportRequest};
 /// Checks for subclasses of `dict`, `list` or `str`.
 ///
 /// ## Why is this bad?
-/// Subclassing `dict`, `list`, or `str` objects can be error prone, use the
-/// `UserDict`, `UserList`, and `UserString` objects from the `collections` module
+/// Built-in types don't consistently use their own dunder methods. For example,
+/// `dict.update()` bypasses `__setitem__`, making inheritance unreliable.
+///
+/// Use the `UserDict`, `UserList`, and `UserString` objects from the `collections` module
 /// instead.
 ///
 /// ## Example
 /// ```python
-/// class CaseInsensitiveDict(dict): ...
+/// class UppercaseDict(dict):
+///     def __setitem__(self, key, value):
+///         super().__setitem__(key.upper(), value)
+///
+/// d = UppercaseDict({"a": 1})
+/// d.update({"b": 2})  # Bypasses __setitem__
+/// print(d)  # {'A': 1, 'b': 2}
```

Comment:
I actually got `{'a': 1, 'b': 2}` when running this. It looks like `dict.__init__` bypasses `__setitem__` too because the `Use instead` example uppercases them both.

```suggestion
/// d = UppercaseDict({"a": 1})
/// d.update({"b": 2})  # Bypasses __setitem__
/// print(d)  # {'a': 1, 'b': 2}
```

Reply:
Thanks for the review, and good catch! I'll update the example to be a bit more accurate 😄